lib/frodo/frodo_helpers.rb

```diff
@@ -7,13 +7,19 @@ def acl
     end
 
     def current_user
-      @frodo_user ||= Frodo::User
-                      .instance(acl)
-                      .frodo_user # NOTE: "User" can be a Client Application or a User
+      @current_user ||= Frodo::User
+                        .instance(acl)
+                        .frodo_user # NOTE: "User" can be a Client Application or a User
     end
 
-    def current_application
-      @current_application ||= Frodo::User.instance(acl).client_application
+    def current_application_name
+      @current_application_name ||= begin
+        if current_user.type == 'users'
```

Comment:
Our apps are type users in gandalf? What type are users then?

Reply:
Our apps are type `client_applications`.  I'm testing to see if it's a `user` type, and, if it is, get the name from the embedded `client_application`.  Otherwise, we know it's a `client_application` so you can just take the name straight from there

Reply:
Ohh I see. And `user` `current_user` is always guaranteed to have a `client_application`?

Reply:
Nm, I know that's true.